qutebrowser/mainwindow/tabwidget.py

```diff
@@ -68,15 +67,15 @@ def __init__(self, win_id, parent=None):
         self.setStyle(TabBarStyle())
         self.setTabBar(bar)
         bar.tabCloseRequested.connect(self.tabCloseRequested)
-        bar.tabMoved.connect(functools.partial(
-            QTimer.singleShot, 0, self.update_tab_titles))
```

Comment:
In order to remove the signal more easily in the subclass, I removed the QTimer; Is that a good idea? Why was it there in the first place, since it doesn't look there's any difference in performance?

Reply:
Looks like I added that in 66d3ec1c081e4a99890f26b0777233f3ae382d7f more than four years ago  - I don't really remember to be honest. Maybe to make sure the tab titles are updated with the new indices (after the move), not the old (before the move) ones?

Reply:
I tested moving and removing tabs without the QTimer and didn't find any problem. There could be some sort of race condition that justifies this, but it doesn't seem so: the Qt5 docs say that the tabMoved signal is "emitted when the tab _has moved_". 

There is a similar pattern (QTimer.singleShot with 0 millis) in many other places in the codebase as well, so the reason why it's there might just be habit or following a pattern

If we're not sure removing it won't be a problem, I can find another workaround (probably saving the partial function or the ID of the connection as an attribute), but the current solution is cleaner.

Reply:
> There is a similar pattern (QTimer.singleShot with 0 millis) in many other places in the codebase as well, so the reason why it's there might just be habit or following a pattern

To be fair, 28 isn't that many in a codebase with some tens of thousands lines of code - for most of them, I know why they're needed right away, for others I'm pretty sure I do :wink:

I'll do a few tests when I get round to reviewing the PR, but I guess if it works it's fine to remove it.

Reply:
That makes sense. Thank you!